services/requester/batch_tx_pool.go

```diff
@@ -156,6 +202,26 @@ func (t *BatchTxPool) processPooledTransactions(ctx context.Context) {
 	}
 }
 
+func (t *BatchTxPool) processIndividualTransactions(ctx context.Context) {
+	defer func() {
+		close(t.txChan)
+	}()
```

Comment:
I think this will cause the node to crash if a transaction is sent during shutdown. 

as a best practice, only close channels when the writer shuts down. Here, that's not possible so it's better to not close it at all.

Reply:
Good catch :100: Removed in https://github.com/onflow/flow-evm-gateway/pull/835/commits/d4aab816729742f3773a71cf72684f2028642c46 .